targets/TARGET_NUVOTON/TARGET_M261/trng_api.cpp

```diff
@@ -0,0 +1,119 @@
+/* 
+ * Copyright (c) 2019-2020 Nuvoton Technology Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if DEVICE_TRNG
+
+#include "cmsis.h"
+#include <limits.h>
+#include "crypto-misc.h"
+#include "hal/trng_api.h"
+#include "platform/mbed_toolchain.h"
+#include "platform/mbed_critical.h"
+#include "platform/mbed_error.h"
+#include "platform/SingletonPtr.h"
+#include "platform/PlatformMutex.h"
+#include "nu_modutil.h"
+
+/* Module init definition: modname, clkidx, clksrc, clkdiv, rstidx, irqnum, misc */
+static const struct nu_modinit_s trng_modinit = {TRNG_0, TRNG_MODULE, 0, 0, TRNG_RST, TRNG_IRQn, NULL};
+
+/* TRNG init counter. TRNG is kept active as it is non-zero. */
+static uint16_t trng_init_counter = 0U;
+
+/* Mutex for synchronizing access to TRNG H/W */
+static SingletonPtr<PlatformMutex> trng_mutex;
```

Comment:
is this needed? Because the upper layer takes care of this - HAL does not by default. 

I assume this TRNG is just one peripheral so this is not taking care of sync in HAL layer when there are multiple users of this peripheral, there's just one trng hal? Or this is connected to secure/non secure thus this lock is needed.

Reply:
@0xc0170 Yes. It is unnecessary.  [mbedtls_mutex](https://github.com/ARMmbed/mbed-os/blob/6240335855e5bac67281c2e122864da53191dd59/features/mbedtls/platform/src/mbed_trng.cpp#L23) has done. Removed it.